test/ChildChainGaugeInjectorV2.js

```diff
@@ -74,13 +78,113 @@ describe('ChildChainGaugeInjector', () => {
             calldata
         );
 
-        await impersonateAccount(ARBI_LDO_WHALE);
-        const whaleSigner = await hre.ethers.getSigner(ARBI_LDO_WHALE);
-        token = new hre.ethers.Contract(ARBI_LDO_ADDRESS, erc20ABI, whaleSigner)
+        await impersonateAccount(TEST_TOKEN_WHALE);
+        const whaleSigner = await hre.ethers.getSigner(TEST_TOKEN_WHALE);
+        token = new hre.ethers.Contract(TEST_TOKEN_ADDRESS, erc20ABI, whaleSigner)
+        tokenDecimals = await token.decimals();
+
+        await token.transfer(owner, 1000e6)
+        console.log(`Balance TEST TOKEN: ${hre.ethers.formatEther(await token.balanceOf(owner))}`);
+
+    });
+
+
+    it("should not allow adding a recipient with amount exceeding MaxInjectionAmount", async function () {
+        const maxInjectionAmount = await injector.MaxInjectionAmount();
+        const exceedingAmount = maxInjectionAmount + hre.ethers.parseEther("1");
+        const gaugeAddress = await gauge.getAddress();
+
+        await expect(
+            injector.addRecipients([gaugeAddress], exceedingAmount, 2, 0)
+        ).to.be.revertedWithCustomError(injector, "ExceedsMaxInjectionAmount");
+    });
+
+    it("should allow adding a recipient with amount equal to MaxInjectionAmount", async function () {
+        const maxInjectionAmount = await injector.MaxInjectionAmount();
+        const gaugeAddress = await gauge.getAddress();
+
+        await expect(
+            injector.addRecipients([gaugeAddress], maxInjectionAmount, 2, 0)
+        ).to.not.be.reverted;
+    });
+
+    it("should update MaxInjectionAmount and affect recipient addition", async function () {
```

Comment:
Nice.  So we test both if a new program will exceed the max, or if the max will be under the current program load. right?